### PR TITLE
feat(terraform): kubernetes image resources support

### LINKED ISSUE #13461: Terraform kubernetes image
### What would you like Renovate to be able to do?

Hello,

I would like renovate to update kubernetes' Terraform definition.
The same way renovate currently update kubernetes yaml file. I don't know if this update should belong to kubernetes manager or terraform manager, but since this could go beyond just kubernetes I guess it's more logical to be handled by terraform manager.

It should match:
`image(\s*)?=(\s*)?(['"]{1})([a-zA-Z0-9-_.:\/]+)['"]`
https://regex101.com/r/IOH952/2

Non exhaustive list of resources:
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/pod
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/pod_v1
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/deployment
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/deployment_v1
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/stateful_set
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/stateful_set_v1
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/daemonset
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/daemon_set_v1
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/job
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/job_v1
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/cron_job
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/replication_controller
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/replication_controller_v1
- ...

### If you have any ideas on how this should be implemented, please tell us here.

I don't know how it's implemented in the kubernetes manager, but I guess a regex should be a fairly simple v1.
But not bulletproof since image could be stored in variable, and could create false positive.

### Is this a feature you are interested in implementing yourself?

No

### LINKED ISSUE #15849: Terraform manager wrongly parse image.
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.69.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab 14.10.2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Terraform manager wrongly parse docker_image depName. This result in a wrong call to the wrong registry
my resource is as such
```
resource "kubernetes_cron_job" "foo" {
...
  spec {
   ...
    job_template {
      ...
      spec {
        ...
        template {
          ...
          spec {
            ...
            container {
              name    = "my-image"
              image   = "my.registry.io/my-repo/my-image:1.0.0@sha256:123456" 
              ....
            }
          }
        }
      }
    }
  }
}
```
In the log I'm seeing.
```
resource "kubernetes_cron_job" "foo" {
...
  spec {
   ...
    job_template {
      ...
      spec {
        ...
        template {
          ...
          spec {
            ...
            container {
              name    = "my-image"
              image   = "my.registry.io/my-repo/my-image:1.0.0@sha256:123456" 
              ....
            }
          } 
        }
      }
    }
  }
}


DEBUG: Dependency my-image has unsupported value my.registry.io/my-repo/my-image:1.0.0@sha256:123456 (repository=....)
DEBUG: getDigest(https://index.docker.io, library/my-image, my.registry.io/my-repo/my-image:1.0.0) (repository=....)
DEBUG: getManifestResponse(https://index.docker.io, library/my-image, my.registry.io/my-repo/my-image:1.0.0) (repository=...)
```

```
       },
       "registryHost": "https://index.docker.io",
       "dockerRepository": "library/my-image",
       "tag": " my.registry.io/my-repo/my-image:1.0.0"
},
       "registryHost": "https://index.docker.io/",
       "dockerRepository": "library/my-image",
       "tag": " my.registry.io/my-repo/my-image:1.0.0"
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

[reproduction repository](https://github.com/vlagorce/renovate-terraform)

### PR DESCRIPTION
**Pull request by Churro**
## Changes

* This PR adds support for container images that are referenced in resources of the [K8s TF provider](https://registry.terraform.io/providers/hashicorp/kubernetes). This provider is covered as a whole, i.e., with all potential and legitimate occurrences.

**No existing tests were changed/rewritten, therefore I don't expect any breaking changes.**

Test repositories (= forking the two minimal repoduction repos from the issues linked below):
* https://github.com/Churro/renovate-terraform1/pull/1
* https://github.com/Churro/renovate-terraform-2/pull/3

## Context

* Closes https://github.com/renovatebot/renovate/issues/15849
* Closes https://github.com/renovatebot/renovate/issues/13461

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/terraform/__fixtures__/kubernetes.tf [hunk 1]
```diff
@@ -0,0 +1,240 @@
+resource "kubernetes_cron_job_v1" "demo" {
```
Comment (secustor) [inline lib/modules/manager/terraform/__fixtures__/kubernetes.tf:1]: Please add an unknown resource.

**Review by secustor — CHANGES_REQUESTED** [review]
Thanks! 

Some small remarks.    
Please also add the new depTypes to the README

File: lib/modules/manager/terraform/extract.ts [hunk 2]
```diff
@@ -34,11 +34,12 @@ const dependencyBlockExtractionRegex = regEx(
 const contentCheckList = [
   'module "',
   'provider "',
+  'image ',
```
[the remark this answers is not in the document]
Comment (Churro) [reply lib/modules/manager/terraform/extract.ts:37]: Thanks, good catch!

File: lib/modules/manager/terraform/__fixtures__/kubernetes.tf (hunk 1 above)
Comment (Churro) [reply lib/modules/manager/terraform/__fixtures__/kubernetes.tf:1]: Done 👍

File: lib/modules/manager/terraform/readme.md [hunk 3]
```diff
@@ -13,6 +13,8 @@ Currently, Terraform supports renovating the following dependencies, where sub p
   - chart repository ( Public and Private )
 - docker\_\*
   - Docker registry ( Public and Private )
+- kubernetes\_\*
+  - Kubernetes registry ( Public and Private )
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:17]:
```suggestion
  - Docker registry ( Public and Private )
```
The reference is a Docker or OCI registry

**Review by secustor — CHANGES_REQUESTED** [review]
small docs fixup, else LGTM

File: lib/modules/manager/terraform/readme.md (hunk 3 above)
Comment (Churro) [reply lib/modules/manager/terraform/readme.md]: You're right, thx 👍

**Review by secustor — DISMISSED** [review]

File: lib/modules/manager/terraform/extract.spec.ts [hunk 4]
```diff
@@ -104,6 +105,13 @@ describe('modules/manager/terraform/extract', () => {
       expect(res).toMatchSnapshot();
     });
 
+    it('extracts kubernetes resources', async () => {
+      const res = await extractPackageFile(kubernetes, 'kubernetes.tf', {});
+      expect(res.deps).toHaveLength(16);
+      expect(res.deps.filter((dep) => dep.skipReason)).toHaveLength(2);
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/manager/terraform/extract.spec.ts:112]: Please use `toMatchObject` and only match required fields

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
The docs look good to go.

`Solves` is not a valid keyword to automatically close a issue on GitHub. Please replace the word `solves` with `Closes` to close the linked issues when the PR gets merged. :wink:

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/terraform/extract.spec.ts (hunk 4 above)
Comment (Churro) [reply lib/modules/manager/terraform/extract.spec.ts]: Done

**Review by viceice — APPROVED** [review]

**Comment by bodinsamuel** [comment]
Thanks a lot for this <3

**Pull request merged**